packages/eslint-plugin-template/src/rules/i18n.ts

```diff
@@ -196,17 +206,27 @@ export default createESLintRule<Options, MessageIds>({
         return;
       }
 
-      if (!isEmpty(customId)) {
-        const sourceSpans = collectedCustomIds.get(customId) ?? [];
-        collectedCustomIds.set(customId, [...sourceSpans, sourceSpan]);
-        return;
-      }
-
       const loc = parserServices.convertNodeSourceSpanToLoc(sourceSpan);
-      context.report({
-        messageId: 'i18nCustomIdOnElement',
-        loc,
-      });
+
+      if (!requireDescription) {
```

Comment:
Perhaps this could be simplified...

Reply:
Yes, I don't think you need the extra `if ... else ...` here. It could be that both customId and description are required and missing, so I assume we'd want to report both.

Reply:
Without the `if ... else ...`, we get test results like the following:
```
    Expected value to strictly be equal to:
      1
    Received:
      2

    Message:
      Should have 1 error but had 2: [
      {
        ruleId: 'i18n',
        severity: 1,
        message: 'Missing custom ID on element. See more at https://angular.io/guide/i18n#manage-marked-text-with-custom-ids',
        line: 2,
        column: 7,
        nodeType: null,
        messageId: 'i18nCustomIdOnElement',
        endLine: 2,
        endColumn: 26
      },
      {
        ruleId: 'i18n',
        severity: 1,
        message: 'Missing i18n description on element. See more at https://angular.io/guide/i18n-common-prepare#i18n-metadata-for-translation',
        line: 2,
        column: 7,
        nodeType: null,
        messageId: 'i18nMissingDescription',
        endLine: 2,
        endColumn: 26
      }
    ]
```

This is because the `checkId` option is `true` by default. Because of this, the two avenues I considered were:
1. Add the conditional in the `handleElement` logic to separate the handling of IDs and other metadata
2. Add in the docs to set `checkId` to `false` if you want to require a description

Option 1 is a lot more user-friendly, and option 2 will become even more problematic if/when we add an option to make the meaning required.

Reply:
(this was addressed offline)